frontend-new/pages/main/index.tsx

```diff
@@ -1,31 +1,39 @@
+import styled from '@emotion/styled'
+import { Box } from '@mui/material'
+import { find } from 'lodash'
 import { useCallback, useMemo, useState } from 'react'
 import useSWR from 'swr'
-import styled from '@emotion/styled'
+
+import { Link, LinkCreate, LinkCreateResponse, User } from 'app/types'
+import { fetcher } from 'app/utils/fetcher'
+
 import { LinkCreationForm } from './components/LinkCreationForm'
 import { LinkList } from './components/LinkList'
 import { ResponseContainer, ResponseType } from './components/ResponseContaner'
 import { Search } from './components/Search'
-import { fetcher } from '../../utils/fetcher'
-import { Link, LinkCreate, LinkCreateResponse } from '../../types'
-import { find } from 'lodash'
 
-const StyledDiv = styled.div`
+const Container = styled.div`
   background-color: #fff;
-  height: calc(100% - 64px);
+  height: calc(100%);
   padding: 64px 200px 0;
-
-  .scrollable {
-    height: calc(100% - 176.5px);
-    overflow: scroll;
-  }
 `
+
+const ScrollableArea = styled.div<{ cut: number }>(({ cut }) => ({
+  height: `calc(100% - ${cut}px)`,
+  overflow: 'scroll',
+}))
+
 interface NotificationState {
   link: Link
   type: ResponseType
   message: string
 }
 
-export default function Home() {
+interface Props {
+  user?: User
+}
+
+export default function Home({ user }: Props) {
   const [notificationState, setNotificationState] = useState<NotificationState>()
```

Comment:
We should extract this logic into a common place. We can begin by moving this into a custom `useLinks` hook

Reply:
just for this component? Or are you meaning to have a hook that can be reused in other components and incapsulates `useContext` inside?